### PR TITLE
YLP-1212

### PR DESCRIPTION
**Pull request by ydrisR**
Integrate shoreline with french eCPS platform.
The goal is to certify french HCP identity

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,13 @@
 
 Shoreline is the module that manages user accounts and authentication.
 
-## 1.8.7 - 2021-01-10 
+## 1.9.0 - 2022-02-17
```
Comment (CedricPerotto) [inline CHANGELOG.md:5]: The 1.8.7 version has been dropped, and the description of changes in 1.9.0 is missing

File: user/api.go [hunk 2]
```diff
@@ -691,6 +753,163 @@ func (a *Api) DeleteUser(res http.ResponseWriter, req *http.Request, vars map[st
 	sendModelAsResWithStatus(res, status.NewStatus(http.StatusForbidden, STATUS_MISSING_ID_PW), http.StatusForbidden)
 }
 
+// @Summary Update an HCP user with an external OAuth ID
```
Comment (CedricPerotto) [inline user/api.go]: Maybe we could separate all these oauth / eCPS specific functions to another file (in order to split a little the massive api.go file).

File: shoreline.go [hunk 3]
```diff
@@ -53,7 +53,15 @@ func main() {
 		logLevel = log.WarnLevel
 	}
 	logger.SetLevel(logLevel)
-	logger.Infof("Starting shoreline service %v\n", version.GetVersion().String())
+
+	// Development mode?
+	appEnv := os.Getenv("APP_ENV")
+	if appEnv == "development" {
+		logger.Info("starting shoreline in development mode")
```
Comment (TimCanu) [inline shoreline.go]: I feel like this log is not necessary as we have a similar one line 64, or am I missing something?

File: user/api.go [hunk 4]
```diff
@@ -691,6 +753,163 @@ func (a *Api) DeleteUser(res http.ResponseWriter, req *http.Request, vars map[st
...
+		return
+	} else if oAuthTokens.Decode(oAuthCookieVal) != nil {
+		a.sendError(res, http.StatusInternalServerError, "Error while decoding Oauth cookie", log, err)
+		return
+	} else if jwtToken, _ := jwt.Parse(oAuthTokens.AuthToken, nil); jwtToken == nil {
+		a.sendError(res, http.StatusInternalServerError, STATUS_ERR_FINDING_USR, log, STATUS_USER_NOT_FOUND, err)
+		return
+	} else {
+		claims := jwtToken.Claims.(jwt.MapClaims)
+		user.FrProId = claims["sub"].(string)
```
Comment (TimCanu) [inline user/api.go]: Can `claims["sub"]` be null? Does it matter?

**Review by TimCanu — COMMENTED** [review]
There is no tests, was it too tricky to do?

File: user/api.go [hunk 5]
```diff
@@ -691,6 +753,163 @@ func (a *Api) DeleteUser(res http.ResponseWriter, req *http.Request, vars map[st
...
+		AuthToken:    tokens.AccessToken,
+		RefreshToken: tokens.RefreshToken,
+	}
+	if cookieVal, err := oidcTokens.Encode(); err != nil {
+		a.logger.Error("Error while encoding JWT token before sending to Blip.", err)
+		a.redirectToBlipError(res, "Internal server error")
+		return
+	} else {
+		rp.CookieHandler().SetCookie(res, "ecps-oidc", cookieVal)
+	}
```
Comment (Nas84) [inline user/api.go]: Maybe we could make this a function, these long 'if else' is difficult to understand

File: shoreline.go (hunk 3 above)
Comment (ydrisR) [reply shoreline.go]: correct...

File: user/api.go (hunk 2 above)
Comment (ydrisR) [reply user/api.go]: :+1:

**Review by TimCanu — APPROVED** [review]

**Pull request merged**